Mysql and Postgres backend tests fail if ran as described in the README.

<!--
Please use this template while reporting a bug and provide as much info as possible.
Questions should be posted to https://community.grafana.com
Use query inspector to troubleshoot issues: https://community.grafana.com/t/using-grafanas-query-inspector-to-troubleshoot-issues/2630
-->

**What happened**:

Running the Golang Postgres and Mysql backend tests as shown in the README, passing the root `pkg` folder:

```
# Run Golang tests using mysql as database - convenient to use /docker/blocks/mysql_tests
GRAFANA_TEST_DB=mysql go test ./pkg/...

# Run Golang tests using postgres as database - convenient to use /docker/blocks/postgres_tests
GRAFANA_TEST_DB=postgres go test ./pkg/...
```

Causes some tests to report as failed due to relations not found in the DB. Taking the postgres tests as an example:

```
--- FAIL: TestAlertRuleExtraction (0.13s)
    extractor_test.go:202: Failed to init test database: Migration failed err: pq: relation "user" does not exist
--- FAIL: TestAlertRuleModel (0.09s)
    rule_test.go:50: Failed to init test database: Migration failed err: pq: relation "user" does not exist
FAIL
FAIL	github.com/grafana/grafana/pkg/services/alerting	0.311s
ok  	github.com/grafana/grafana/pkg/services/alerting/conditions	0.075s
ok  	github.com/grafana/grafana/pkg/services/alerting/notifiers	0.088s
?   	github.com/grafana/grafana/pkg/services/annotations	[no test files]
...............................................................................
79 total assertions
```

**What you expected to happen**:

All the Golang tests to pass.

**How to reproduce it (as minimally and precisely as possible)**:

* Checkout v6.2.5 Grafana - `git checkout v6.2.5`
* Setup the test docker-compose, in `devenv`:
    * `./create_docker_compose.sh postgres_tests`
    * `docker-compose up`
* Clean the Golang test cache: `go clean -testcache`
* Run the Golang tests passing in the `pkg` root directory: `GRAFANA_TEST_DB=postgres go test ./pkg/...`

**Anything else we need to know?**:

I've not done the digging on why running go test from the root `pkg` directory doesn't work but given that the circle CI runs it in a different way suggests that somebody already knows about this behaviour.

In the circle ci postgres script, the tests are invoked in each individual package with a loop:

```
time for d in $(go list ./pkg/...); do
 exit_if_fail go test -tags=integration $d
done
```

Running a similar command makes the tests pass:

```for d in $(go list ./pkg/...); do GRAFANA_TEST_DB=postgres go test $d; done```


So changing the README to the following (with a note why running using the root `pkg` directory doesn't work?) should be fine:

```
# Run Golang tests using mysql as database - convenient to use /docker/blocks/mysql_tests
`for d in $(go list ./pkg/...); do GRAFANA_TEST_DB=mysql go test $d; done

# Run Golang tests using postgres as database - convenient to use /docker/blocks/postgres_tests
`for d in $(go list ./pkg/...); do GRAFANA_TEST_DB=postgres go test $d; done
```

**Environment**:
- Grafana version: 6.2.5 
- Data source type & version: NA
- OS Grafana is installed on: Centos 7
3.10.0-862.14.4.el7.x86_64
- User OS & Browser: NA
- Grafana plugins: NA
- Others: NA

--- a/README.md
+++ b/README.md
@@ -147,12 +147,34 @@ Writing & watching frontend tests
 ```bash
 # Run Golang tests using sqlite3 as database (default)
 go test ./pkg/...
+```
 
-# Run Golang tests using mysql as database - convenient to use /docker/blocks/mysql_tests
-GRAFANA_TEST_DB=mysql go test ./pkg/...
+##### Running the MySQL or Postgres backend tests:
 
-# Run Golang tests using postgres as database - convenient to use /docker/blocks/postgres_tests
-GRAFANA_TEST_DB=postgres go test ./pkg/...
+Run these by setting `GRAFANA_TEST_DB` in your environment.
+
+- `GRAFANA_TEST_DB=mysql` to test MySQL
+- `GRAFANA_TEST_DB=postgres` to test Postgres
+
+Follow the instructions in `./devenv` to spin up test containers running the appropriate databases with `docker-compose`
+- Use `docker/blocks/mysql_tests` or `docker/blocks/postgres_tests` as appropriate
+
+```bash
+# MySQL
+# Tests can only be ran in one Go package at a time due to clashing db queries. To run MySQL tests for the "pkg/services/sqlstore" package, run:
+GRAFANA_TEST_DB=mysql go test ./pkg/services/sqlstore/...
+
+# Or run all the packages using the circle CI scripts. This method will be slower as the scripts will run all the tests, including the integration tests.
+./scripts/circle-test-mysql.sh 
+```
+
+```bash
+# Postgres
+# Tests can only be ran in one Go package at a time due to clashing db queries. To run Postgres tests for the "pkg/services/sqlstore" package, run:
+GRAFANA_TEST_DB=postgres go test ./pkg/services/sqlstore/...
+
+# Or run all the packages using the circle CI scripts. This method will be slower as the scripts will run all the tests, including the integration tests.
+./scripts/circle-test-postgres.sh
 ```
 
 #### End-to-end

